File: PracticasProgramacion/Practica_06/cpp/Calculadora.cpp
#include "Calculadora.h"
#include <stdexcept>

double Calculadora::sumar(double a, double b) const { return a + b; }
double Calculadora::restar(double a, double b) const { return a - b; }
double Calculadora::multiplicar(double a, double b) const { return a * b; }
double Calculadora::dividir(double a, double b) const {
    if (b == 0) throw std::runtime_error("Division por cero");
    return a / b;
}

File: PracticasProgramacion/Practica_03/python/calculadora.py
"""
Calculadora: cuatro operaciones aritméticas.
División lanza ValueError si divisor es cero.
"""


class Calculadora:
    @staticmethod
    def sumar(a: float, b: float) -> float:
        return a + b

    @staticmethod
    def restar(a: float, b: float) -> float:
        return a - b

    @staticmethod
    def multiplicar(a: float, b: float) -> float:
        return a * b

    @staticmethod
    def dividir(a: float, b: float) -> float:
        if b == 0:
            raise ValueError("División por cero")
        return a / b
